### PR TITLE
Add an internal route for bin/rails notes

### PR DESCRIPTION
**Pull request by deepakmahakale**
### Motivation / Background

Looking for early feedback on notes

Inspired by the `rails/info/routes` route I am suggesting a `rails/info/notes` internal route.

This will be same as doing 

```sh
$ bin/rails notes

app/controllers/posts_controller.rb:
  * [ 9] [TODO] Move this logic to a concern
  * [18] [FIXME] Refactor this method

app/models/post.rb:
  * [ 2] [TODO] Refactor this validation
```

### Detail

Adding an internal route for the `bin/rails notes` so we can check the notes on UI.

### Additional information

**Option 1:**

Screenshot 2023-09-12 at 19 40 07

**Option 2:**

Screenshot 2023-09-12 at 19 36 03

### Checklist

Before submitting the PR make sure the following are checked:

* [x] This Pull Request is related to one change. Changes that are unrelated should be opened in separate PRs.
* [x] Commit message has a detailed description of what changed and why. If this PR fixes a related issue include it in the commit message. Ex: `[Fix #issue-number]`
* [ ] Tests are added or updated if you fix a bug or add a feature.
* [ ] CHANGELOG files are updated for the changed libraries if there is a behavior change or additional feature. Minor bug fixes and documentation changes should not be included.

### CODE REVIEW

**Comment by deepakmahakale** [comment]
@rafaelfranca thoughts on this?

**Comment by willianveiga** [comment]
Did you start a discussion on the [rails-core discussion board](https://discuss.rubyonrails.org/c/rubyonrails-core) about this feature as pointed out [here](https://edgeguides.rubyonrails.org/contributing_to_ruby_on_rails.html#what-about-feature-requests-questionmark)?

**Comment by vipulnsward** [comment]
@deepakmahakale this needs a changelog

**Comment by deepakmahakale** [comment]
@vipulnsward raised a pr - https://github.com/rails/rails/pull/52115

File: railties/lib/rails/templates/rails/info/notes.html.erb [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    <th>File Name</th>
+    <th class="line-no">Line No.</th>
+    <th class="tag">Tag</th>
+    <th>Description</th>
+  </thead>
+  <tbody>
+    <% @annotations.each do |file, annotations| %>
+      <% annotations.each.with_index do |annotation, index| %>
+        <tr>
+          <% if index == 0 %>
```
Comment (justinko) [inline railties/lib/rails/templates/rails/info/notes.html.erb:53]: `index.zero?`

**Pull request merged**